src/librustc_mir/interpret/machine.rs

```diff
@@ -120,6 +120,16 @@ pub trait Machine<'mir, 'tcx>: Sized {
     /// Whether memory accesses should be alignment-checked.
     const CHECK_ALIGN: bool;
 
+    /// Borrow the current thread's stack.
```

Comment:
Please move these down at least below `enforce_validity`... or even better, group all the stack methods together (we already have `stack_push`/`stack_pop`).